extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultService.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.vault;
+
+import java.util.Map;
+
+import io.quarkus.vault.service.dto.VaultDynamicCredentials;
+
+public interface VaultService {
+
+    String DEFAULT_VAULT_SECRET_KEY = "password";
+
+    String getToken();
+
+    Map<String, String> readSecret(String path);
+
+    VaultDynamicCredentials getDynamicCredentials(String dbrole);
```

Comment:
The parameter should probably be called just the `role` (assuming that it is necessarily part of the DB flow)

Reply:
it is for db. probably VaultDynamicCredentials should be renamed into VaultDynamicDbCredentials

Reply:
Yes, that would be consistent with the other DB specific class

Reply:
I have renamed a few things:

1) be consistent with db feature

 - VaultDynamicCredentials => VaultDynamicDbCredentials
 - VaultCreds => VaultDbCreds
 - VaultCredsData => VaultDbCredsData
 - VaultService.getDynamicCredentials(String) => VaultService.getDynamicDbCredentials(String)
 - VaultDynamicCredentials => VaultDynamicDbCredentials
 - VaultBaseProxy.generateCredentials() => VaultBaseProxy.generateDbCredentials()

2) internal stuff

 - VaultDynamicDbSecretService => VaultDynamicDbSecretManager
 - VaultAuthService => VaultAuthManager

3) agroal stuff

 - io.quarkus.agroal.runtime.vault.VaultDynamicPassword => io.quarkus.agroal.runtime.vault.AgroalVaultDynamicPassword